recipe/session/recipe.go

```diff
@@ -187,7 +187,10 @@ func (r *Recipe) handleError(err error, req *http.Request, res http.ResponseWrit
 	if defaultErrors.As(err, &errors.UnauthorizedError{}) {
 		supertokens.LogDebugMessage("errorHandler: returning UNAUTHORISED")
 		unauthErr := err.(errors.UnauthorizedError)
-		return true, r.Config.ErrorHandlers.OnUnauthorised(err.Error(), unauthErr.ClearCookies == nil || *unauthErr.ClearCookies, req, res)
+		if unauthErr.ClearCookies == nil || *unauthErr.ClearCookies {
+			clearSessionFromCookie(r.Config, res)
```

Comment:
do a log before clearSessionFromCookie -> `supertokens.LogDebugMessage("getSession: Clearing cookies because of UNAUTHORISED response")`